### PR TITLE
rebuild everything

### PR DESCRIPTION
**Pull request by KillWolfVlad**
1) rebuild everything with latest electron version
2) enjoy on Arch or Manjaro

P.S. @johnhaley81 didn't help me :angry:

### CODE REVIEW

**Comment by johnhaley81** [comment]
https://github.com/KillWolfVlad/GitKraken-Pro-AUR/issues/2

🤔

File: PKGBUILD [hunk 1]
```diff
@@ -28,16 +30,27 @@ md5sums=('a6dcbb4979b10ea7a865ffa742de8fa2'
...
+    install -d ${pkgdir}/opt
+    cp -R ${srcdir}/gitkraken ${pkgdir}/opt/gitkraken
 
-    find "$pkgdir"/opt/gitkraken/ -type f -exec chmod 644 {} \;
-    chmod 755 "$pkgdir"/opt/gitkraken/gitkraken
+    find ${pkgdir}/opt/gitkraken/ -type f -exec chmod 644 {} \;
+    chmod 755 ${pkgdir}/opt/gitkraken/gitkraken
 
-    install -d "$pkgdir"/usr/bin
+    install -d ${pkgdir}/usr/bin
```
Comment (MarcinWieczorek) [inline PKGBUILD]: You are not supposed to remove the quotes.

File: PKGBUILD [hunk 2]
```diff
@@ -6,15 +6,17 @@
...
 
 pkgname=gitkraken
-pkgrel=1
+pkgrel=2
 pkgver=2.5.0
+nodegitver=0.18.3
+electronver=1.2.8
 pkgdesc="The intuitive, fast, and beautiful cross-platform Git client."
-url="http://www.gitkraken.com/"
+url="https://www.gitkraken.com/"
```
Comment (MarcinWieczorek) [inline PKGBUILD]: Why? Their SSL is valid.

File: PKGBUILD [hunk 3]
```diff
@@ -6,15 +6,17 @@
...
 provides=('gitkraken')
 arch=('x86_64')
 license=('custom')
-depends=('gtk2' 'nss' 'libxtst' 'libgnome-keyring' 'gconf' 'alsa-lib' 'libcurl-openssl-1.0' 'libxss')
-makedepends=()
+depends=('gtk2' 'nss' 'libxtst' 'libgnome-keyring' 'gconf' 'alsa-lib' 'libxss')
+makedepends=('nodejs>=7.7.3' 'npm>=4.5.0')
```
Comment (MarcinWieczorek) [inline PKGBUILD]: Is this all a troll or I am too stupid to understand the logic behind it?

File: PKGBUILD [hunk 4]
```diff
@@ -28,16 +30,27 @@ md5sums=('a6dcbb4979b10ea7a865ffa742de8fa2'
...
          'e9ba6663e5f1f92cae31beb0074e8c6c'
-         '4c7a550e9abee9d2c4ab6046736e9113')
+         '6c504da03a5e7068a29a9618aa29a994')
+
+build() {
+    cd ${srcdir}
+    mkdir nodegit && cd "$_"
+    npm i nodegit@${nodegitver}
+    cd node_modules/nodegit
+    HOME=./.electron-gyp node-gyp rebuild --target=${electronver} --arch=x64 --dist-url=https://atom.io/download/electron
```
Comment (MarcinWieczorek) [inline PKGBUILD]: You can't do such things with ArchLinux.

File: PKGBUILD (hunk 2 above)
Comment (KillWolfVlad) [inline PKGBUILD]: `https` is better
image

File: PKGBUILD (hunk 3 above)
Comment (KillWolfVlad) [inline PKGBUILD]: better to use latest versions

File: PKGBUILD (hunk 1 above)
Comment (KillWolfVlad) [inline PKGBUILD]: didn't know about it

File: PKGBUILD (hunk 4 above)
Comment (KillWolfVlad) [inline PKGBUILD]: What is the difference between `make && make install` and `node-gyp rebuild`?

File: PKGBUILD (hunk 2 above)
Comment (MarcinWieczorek) [inline PKGBUILD]: My fault, I thought It's the other way around. Reading diff is not for everyone!

**Comment by MarcinWieczorek** [comment]
What's the point in committing the same patch again? It's already on master, you're supposed to resolve merge conflicts as easy as possible.

**Comment by KillWolfVlad** [comment]
You can apply only patch without bad commits

**Comment by adalbertobrant** [comment]
Hi , it is not working ! :( I think it is related ...Could you give me a help ?
gitkraken-error

**Comment by KillWolfVlad** [comment]
@adalbertobrant see [this](https://github.com/Azd325/gitkraken/issues/38#issuecomment-301273394) comment

**Comment by KillWolfVlad** [comment]
All changes was included into [master-pro](https://github.com/KillWolfVlad/GitKraken-Pro-AUR) branch

File: gitkraken.sh [hunk 5]
```diff
@@ -1,3 +1,3 @@
-#!/bin/sh
+#!/usr/bin/env sh
```
Comment (LC43) [inline gitkraken.sh]: why is this necessary?

File: PKGBUILD [hunk 6]
```diff
@@ -44,3 +88,39 @@ package() {
     install -D -m644 "./GitKraken.desktop" "${pkgdir}/usr/share/applications/GitKraken.desktop"
     install -D -m644 "./gitkraken.png" "${pkgdir}/usr/share/pixmaps/gitkraken.png"
 }
+
```
Comment (LC43) [inline PKGBUILD]: and what is the purpose of this?

File: PKGBUILD [hunk 7]
```diff
@@ -6,36 +6,80 @@
...
+    rebuild_node nodegit nodegit "$_nodegitver"
+    rebuild_node nsfw nsfw "$_nsfwver"
+    rebuild_node pathwatcher pathwatcher "$_pathwatcherver"
+    rebuild_node runas runas "$_runasver"
+    rebuild_node segfault-handler segfault-handler "$_segfaulthandlerver"
+    asar p "$srcdir"/app "$srcdir"/app.asar
+}
 
 package() {
+    cd "$srcdir"
```
Comment (LC43) [inline PKGBUILD]: no point in cd inside a script

**Comment by LC43** [comment]
Hi, i also have this problem, thanks for your solution.

Can you review the changes you made? seem to do more than needed.

File: PKGBUILD (hunk 6 above)
Comment (KillWolfVlad) [inline PKGBUILD]: for beauty :sunglasses:

File: gitkraken.sh (hunk 5 above)
Comment (KillWolfVlad) [inline gitkraken.sh]: for better portability

File: PKGBUILD (hunk 7 above)
Comment (KillWolfVlad) [inline PKGBUILD]: fixed

**Comment by MarcinWieczorek** [comment]
I think there's no point changing the shebang. `/usr/bin/bash` is valid for everyone as it is provided by the `bash` package which is a part of the base group. Remember that this is a script for makepkg which is useless outside the ArchLinux distribution. Also, /bin is a symlink to /usr/bin, this might cause problems, but not in the case of shebangs.
Also remember that check/prepare/build/package etc always points to $srcdir so cd'ing is useless.
Beauty is an important part of writing a script imho. But the most critical thing in code style is a consistent use of braces in variable names. You're supposed to use either `$pkgdir` or `${pkgdir}`, but never mix them (same goes for `"$pkgdir"/bin/` and `"$pkgdir/bin/"` unless you're using wildcards). There's no point forcing the `$srcdir` path in install command, well, I tend to do that but I never change it when modifying/updating a package. Just leave it as it is.
Also, what is that ASCII gitkraken logo for? Beauty?

**Comment by KillWolfVlad** [comment]
@MarcinWieczorek 

> Just leave it as it is.

OK

> Also, what is that ASCII gitkraken logo for? Beauty?

Yes. This logo is easter egg from @Axosoft. You can see them if launch `gitkraken --nexxuz`

**Comment by MarcinWieczorek** [comment]
Please treat the ArchLinux environment maturely. I don't see a reason to have ASCII art even in the git history.

File: gitkraken.sh (hunk 5 above)
Comment (LC43) [inline gitkraken.sh]: my arch system doesn't use that. Like @MarcinWieczorek said, /bin/sh ( /usr/bin/sh ) is the default.

File: PKGBUILD (hunk 7 above)
Comment (LC43) [inline PKGBUILD]: thanks!

**Comment by KillWolfVlad** [comment]
All changes was included into [master-pro](https://github.com/KillWolfVlad/GitKraken-Pro-AUR) branch

**Comment by MarcinWieczorek** [comment]
We don't really care as we support free software instead of breaking the laws of copyright.
